src/go/k8s/controllers/redpanda/console_controller.go

```diff
@@ -194,6 +206,46 @@ func (r *ConsoleReconciler) createServiceAccount(ctx context.Context, console *r
 	return nil
 }
 
+func (r *ConsoleReconciler) createSASLUser(ctx context.Context, cluster *redpandav1alpha1.Cluster, console *redpandav1alpha1.Console, log logr.Logger) (username, password string, err error) {
+	consoleSu := resources.NewSuperUsers(r.Client, console, r.RuntimeScheme, resources.ScramConsoleUsername, resources.ConsoleSuffix, log)
+	if err := consoleSu.Ensure(ctx); err != nil {
+		return username, password, fmt.Errorf("ensuring console su secret: %w", err)
+	}
+
+	headlessSvc := resources.NewHeadlessService(r.Client, cluster, r.RuntimeScheme, nil, log)
+	clusterSvc := resources.NewClusterService(r.Client, cluster, r.RuntimeScheme, nil, log)
+	fqdn := headlessSvc.HeadlessServiceFQDN(r.clusterDomain)
+	pki := certmanager.NewPki(r.Client, cluster, fqdn, clusterSvc.ServiceFQDN(r.clusterDomain), r.RuntimeScheme, log)
+
+	adminTLSConfigProvider := pki.AdminAPIConfigProvider()
+	adminAPI, err := r.AdminAPIClientFactory(ctx, r, cluster, fqdn, adminTLSConfigProvider)
+	if err != nil && errors.Is(err, &adminutils.NoInternalAdminAPI{}) {
```

Comment:
Shouldn't it return an error? I've seen a similar reasoning in the main redpanda loop, but can't get it.

Reply:
I guess the reasoning there is that the loop will retry anyway, but I don't know if the same is applicable here.

Reply:
Thanks! I overlooked it. This is updated.